### DISCUSSION TITLE
Way to create new pseudo language

### DISCUSSION THREAD

**Discussion by elbiazo** [Vector35/binaryninja-api#3484] [API] [answered]
Is there a way to convert some HLIL into different Text? For example, maybe I want to convert `somefunc(class_obj, funcA)` into `[class_obj::somefunc funcA` for some reason dynamically from python console. Also maybe even convert HLIL into different HLIL dynamically?

**Comment by psifertex** [comment, marked as answer]
So [workflows](https://github.com/Vector35/binaryninja-api/tree/dev/examples/workflows) let you rewrite HLIL, but the only way to make a new IL level would be to create a totally new view. Additionally, workflows can't be triggered on-demand. So the short explanation is there's no easy way to do it now and you'd have to either create a new custom view that re-created the existing IL features you wanted which is a lot of work.
